src/change_stream.ts

```diff
@@ -893,14 +941,17 @@ export class ChangeStream<
     const stream = this[kCursorStream] ?? cursor.stream();
     this[kCursorStream] = stream;
     stream.on('data', change => {
+      this.timeoutContext?.refresh();
```

Comment:
When we refresh/clear the context here - what are we hoping to time out?  If we've entered this listener, we already have received a change event from the underlying cursor's stream.  The processing of events (this function) is synchronous, and there is nothing that waits on the context.

Reply:
Good catch, the timeout context here is getting refreshed and cleared by the underlying cursor and any timeouts happening there get bubbled up as error events.